share/shwap/p2p/shrex/shrexeds/client.go

```diff
@@ -157,6 +157,30 @@ func (c *Client) doRequest(
 	}
 }
 
+// readEDS reads EDS data with context cancellation support.
+func (c *Client) readEDS(
+	ctx context.Context, stream network.Stream, root *share.AxisRoots,
+) (*eds.Rsmt2D, error) {
+	type result struct {
+		eds *eds.Rsmt2D
+		err error
+	}
+	resultCh := make(chan result, 1)
```

Comment:
buffer is not needed

Reply:
I am thinking of the case 
t= 0 goroutine is fired
t=2 context is cancelled  and function has returned
t =4 it tries to send using the ch
select is not waiting to receive the value from ch. In this case goroutine is blocked I think

Reply:
Right, handling context on write side is missing